app/include/zmk/events/keycode-state-changed.h

```diff
@@ -7,22 +7,34 @@
 #pragma once
 
 #include <zephyr.h>
+#include <dt-bindings/zmk/keys.h>
 #include <zmk/event-manager.h>
 
 struct keycode_state_changed {
     struct zmk_event_header header;
     u8_t usage_page;
     u32_t keycode;
+    u8_t mods;
     bool state;
 };
 
 ZMK_EVENT_DECLARE(keycode_state_changed);
 
-inline struct keycode_state_changed *create_keycode_state_changed(u8_t usage_page, u32_t keycode,
-                                                                  bool state) {
+static inline struct keycode_state_changed *create_keycode_state_changed(u8_t usage_page, u32_t keycode, bool state) {
     struct keycode_state_changed *ev = new_keycode_state_changed();
     ev->usage_page = usage_page;
-    ev->keycode = keycode;
+    // this is ugly and after merging https://github.com/zmkfirmware/zmk/pull/300
+    // should move into the keycode_state_changed_from_encoded method
+    // discuss: should we keep LCTL and MOD_LCTL even though it causes this ugly code?
```

Comment:
I don't like this code here, but I'm not sure it's better to in hid.c.

Reply:
I think one good reason to remove this from here is to help keep clearly distinct the modifiers that are being held/applied because they are being asked for in addition to a different keycode, and a modifier that truly is just a modifier being pressed/released.

That *context* I think is important to preserve, because we'll need to when we work to solve the more nuanced handling of modifiers and when we release them because of other key presses, etc.

Thoughts?

Reply:
Agreed, that's a neat way to separate the implicit/explicit modifiers.